app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarkerInformationPopover.tsx

```diff
@@ -0,0 +1,440 @@
+      currentExecution.application,
+      currentExecution.pipelineConfigId,
+    );
+
+    // save this for rendering pipelines
+    this.allExecutions.push({
+      execution: currentExecution,
+      stageId,
+      stageIndex,
+    });
+
+    // get the child execution aka clicking View Pipeline Details
+    const childExecution = await this.informationState.getExecution(currentStage.masterStage.context.executionId);
+    const childTerminalStage = childExecution.stageSummaries.find((stage: IExecutionStageSummary, index: number) => {
+      if (stage.status.toLocaleLowerCase() === 'terminal') {
+        stageIndex = index;
+      }
+
+      return stage.status.toLowerCase() === 'terminal';
+    });
+    const childTerminalPipelineStage = childExecution.stageSummaries.find(
+      (stage: IExecutionStageSummary) => stage.status.toLowerCase() === 'terminal' && stage.type === 'pipeline',
+    );
+    // get the current configuration for this execution
+    const childPipelineConfig = await this.informationState.getPipelineConfig(
+      childExecution.application,
+      childExecution.pipelineConfigId,
+    );
+
+    if (childExecution && !childTerminalPipelineStage) {
+      this.childExecution = childExecution;
+      this.childPipelineConfig = childPipelineConfig;
+
+      // save this for rendering pipelines
+      this.allExecutions.push({
+        execution: childExecution,
+        stageId,
+        stageIndex,
+      });
+    }
+
+    if (childTerminalPipelineStage) {
+      this.getPipelineLink(childTerminalPipelineStage.id, childExecution.id);
+      this.childTerminalPipelineStage = childTerminalPipelineStage;
+    } else {
+      // now that we are complete let's fix up the allExecutions array
+      // we are using allExecutions as a breadcrumb so reverse them then pop the first one since there user is already at the first one
+      this.allExecutions.reverse();
+      this.allExecutions.pop();
+
+      this.setState({
+        pipelineConfigDetails: this.childPipelineConfig || currentPipelineConfig,
+        executionDetails: this.childExecution || currentExecution,
+        failedInApplication: currentStage.masterStage.context.application,
+        link: `/#/applications/${currentStage.masterStage.context.application}/executions/details/${currentStage.masterStage.context.executionId}?stage=${this.allExecutions[0].stageIndex}&step=0`,
+        stageDetails: childTerminalStage || this.childTerminalPipelineStage || currentStage,
+      });
+    }
+  };
+
+  private windowCanFit = () => {
+    const { left, width } = this.props.target.getBoundingClientRect();
+    return this.popoverHalfWidth + left + width - window.innerWidth;
+  };
+
+  private showPipelineGraph = () => {
+    this.setState({ showPipelineGraph: true });
+  };
+
+  private hidePipelineGraph = () => {
+    this.setState({ showPipelineGraph: false });
+  };
+
+  private getMaxHeight = (top: number) => {
+    const height = window.innerHeight - top - 100;
```

Comment:
400 is half width of the popover, bootstrap sets these as a max of 400 i bumped to using 800 since there are a lot of stages and data being displayed, the 100 is just to make sure if you open the popover near the bottom (assuming last execution of last execution group) it won't overly extended the entire page.

Reply:
Resolving as this comment is no longer relevant now that you've refactored to use a `Modal`. 👍